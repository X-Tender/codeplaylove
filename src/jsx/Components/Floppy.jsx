import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Floppy = ({ cover, to, children }) => (
	<Link className="floppy" title={children} to={to}>
		<img className="floppy__disc" src="assets/img/games/floppy.svg" />
		<img className="floppy__cover" src={cover} />
	</Link>
);

Floppy.propTypes = {
	cover: PropTypes.string.isRequired,
	to: PropTypes.string.isRequired,
	children: PropTypes.string.isRequired,
};

export default Floppy;
