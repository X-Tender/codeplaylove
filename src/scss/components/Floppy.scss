.floppy {
	position: relative;
	transition: transform 250ms ease;

	&:after {
		content: '';
		width: calc(100% - 50px);
		height: calc(100% - 45px);
		//background-color: rgba(#29465f, 0.51);
		box-shadow: 0 5px 15px 15px rgba(#29465f, 0.51);
		border-radius: 3%;
		display: block;
		left: 25px;
		top: 25px;
		position: absolute;
		z-index: 0;
	}

	&:hover {
		transform: scale(1.1);
		z-index: 10;
	}

	&__disc {
		object-fit: contain;
		position: relative;
		z-index: 1;
	}
	&__cover {
		z-index: 2;
		object-fit: cover;
		position: absolute;
		width: 72.5%;
		height: 52.5%;
		top: 43.2%;
		left: 13.8%;
		border-radius: 3%;
	}
}

.floppy-container {
	grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
	grid-gap: 30px;
}
