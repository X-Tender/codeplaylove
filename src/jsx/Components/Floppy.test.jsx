import React from 'react';
import { shallow } from 'enzyme';
import Floppy from './Floppy';

describe('Floppy', () => {
	const cover = 'uploads/image.jpg';
	const to = '/my-game';
	const children = 'My Game';

	const comp = shallow(
		<Floppy cover={cover} to={to}>
			{children}
		</Floppy>
	);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('renders the correct props', () => {
		expect(comp.find('.floppy').prop('title')).toEqual(children);
		expect(comp.find('.floppy').prop('to')).toEqual(to);
		expect(comp.find('.floppy__cover').prop('src')).toEqual(cover);
	});
});
