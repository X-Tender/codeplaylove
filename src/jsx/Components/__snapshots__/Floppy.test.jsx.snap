// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floppy renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Floppy
    cover="uploads/image.jpg"
    to="/my-game"
  >
    My Game
  </Floppy>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          className="floppy__disc"
          src="assets/img/games/floppy.svg"
        />,
        <img
          className="floppy__cover"
          src="uploads/image.jpg"
        />,
      ],
      "className": "floppy",
      "replace": false,
      "title": "My Game",
      "to": "/my-game",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "floppy__disc",
          "src": "assets/img/games/floppy.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "floppy__cover",
          "src": "uploads/image.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            className="floppy__disc"
            src="assets/img/games/floppy.svg"
          />,
          <img
            className="floppy__cover"
            src="uploads/image.jpg"
          />,
        ],
        "className": "floppy",
        "replace": false,
        "title": "My Game",
        "to": "/my-game",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "floppy__disc",
            "src": "assets/img/games/floppy.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "floppy__cover",
            "src": "uploads/image.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
